cmd/kubeadm/app/discovery/discovery.go

```diff
@@ -20,14 +20,68 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/discovery/file"
+	"k8s.io/kubernetes/cmd/kubeadm/app/discovery/https"
+	"k8s.io/kubernetes/cmd/kubeadm/app/discovery/token"
 	kubenode "k8s.io/kubernetes/cmd/kubeadm/app/node"
 	tokenutil "k8s.io/kubernetes/cmd/kubeadm/app/util/token"
 )
 
+func Check(discURL, discFile, discToken string) (string, error) {
```

Comment:
This code should be in the API validation when you've added the properties there